src/pykeen/losses.py

```diff
@@ -295,6 +393,53 @@ def __init__(
         else:
             self.margin_activation = margin_activation
 
+    def process_slcwa_scores(
+        self,
+        positive_scores: torch.FloatTensor,
+        negative_scores: torch.FloatTensor,
+        label_smoothing: Optional[float] = None,
+        batch_filter: Optional[torch.BoolTensor] = None,
+        num_entities: Optional[int] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # Sanity check
+        if label_smoothing:
+            raise UnsupportedLabelSmoothingError(self)
+
+        # prepare for broadcasting, shape: (batch_size, 1, 3)
+        positive_scores = positive_scores.unsqueeze(dim=1)
+
+        if batch_filter is not None:
+            # negative_scores have already been filtered in the sampler!
```

Comment:
So can this logic be removed from the process scores functions? Right now the batch_filter isn't used for anything else

Reply:
used in NSSA loss, can't avoid